src/content/en/updates/2017/06/object-rest-spread.md

```diff
@@ -0,0 +1,84 @@
+project_path: /web/_project.yaml
+book_path: /web/updates/_book.yaml
+description: Object rest & spread properties
+
+{# wf_updated_on: 2017-06-05 #}
+{# wf_published_on: 2017-06-05 #}
+{# wf_tags: javascript #}
+
+{% include "web/tools/chrome-devtools/_shared/styles.html" %}
+
+# Object rest & spread properties {: .page-title }
+
+{% include "web/_shared/contributors/mathiasbynens.html" %}
+
+Good ol’ ECMAScript 2015 introduced _rest elements_ for array destructuring
+assignment and _spread elements_ for array literals.
+
+    // Rest elements for array destructuring assignment:
+    const primes = [2, 3, 5, 7, 11];
+    const [first, second, ...rest] = primes;
+    console.log(first); // 2
+    console.log(second); // 3
+    console.log(rest); // [5, 7, 11]
+
+    // Spread elements for array literals:
+    const primesCopy = [first, second, ...rest];
+    console.log(primesCopy); // [2, 3, 5, 7, 11]
+
+These ES2015 features have been supported since Chrome 46 and Chrome 47,
+respectively.
+
+[A stage 3 proposal](https://github.com/tc39/proposal-object-rest-spread)
+enables rest and spread properties for object literals, too.
+
+    // Rest properties for object destructuring assignment:
+    const person = {
+        firstName: 'Sebastian',
+        lastName: 'Markbåge',
+        country: 'USA',
+        state: 'CA',
+    };
+    const { firstName, lastName, ...rest } = person;
+    console.log(firstName); // Sebastian
+    console.log(lastName); // Markbåge
+    console.log(rest); // { country: 'USA', state: 'CA' }
+
+    // Spread properties for object literals:
+    const personCopy = { firstName, lastName, ...rest };
+    console.log(personCopy);
+    // { firstName: 'Sebastian', lastName: 'Markbåge', country: 'USA', state: 'CA' }
+
+Spread properties offer a more elegant alternative to
+[`Object.assign()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign)
+in many situations:
+
+    // Shallow-clone an object:
+    const data = { x: 42, y: 27, label: 'Treasure' };
+    // The old way:
+    const clone1 = Object.assign({}, data);
+    // The new way:
+    const clone2 = { ...data };
+    // Either results in:
+    // { x: 42, y: 27, label: 'Treasure' }
+
+    // Merge two objects:
+    const defaultSettings = { logWarnings: false, logErrors: false };
+    const userSettings = { logErrors: true };
+    // The old way:
+    const settings1 = Object.assign({}, defaultSettings, userSettings);
+    // The new way:
+    const settings2 = { ...defaultSettings, ...userSettings };
+    // Either results in:
+    // { logWarnings: false, logErrors: true }
+
+However, there are [some subtle
+differences](http://2ality.com/2016/10/rest-spread-properties.html#spread-defines-properties-objectassign-sets-them)
+in how spreading handles setters.
```

Comment:
I think we should highlight the differences to `Object.assign` more clearly here.